### PR TITLE
1.12-rc1 cherry-pick request: Add mode_override to the TPU embedding enqueue ops. 

### PR DESCRIPTION
**Pull request by tatatodd**
This allows the mode to be overridden at runtime allowing dynamic switching between inference and training modes. Not fully implemented yet.

PiperOrigin-RevId: 215325071

### CODE REVIEW

**Review by saeta — APPROVED** [review]
LGTM, but I defer to @jhseu if he has any issues.

**Review by jhseu — APPROVED** [review]
(I already reviewed this change internally and agree it should be in 1.12)

**Review by frankchn — APPROVED** [review]

**Comment by tatatodd** [comment]
(Added @george-kurian just as an FYI)

**Pull request merged**
